internal/cephfs/nodeserver.go

```diff
@@ -73,11 +73,40 @@ func getCredentialsForVolume(
 	return cr, nil
 }
 
+func (ns *NodeServer) getVolumeOptions(
+	ctx context.Context,
+	volID fsutil.VolumeID,
+	volContext,
+	volSecrets map[string]string,
+) (*core.VolumeOptions, error) {
+	volOptions, _, err := core.NewVolumeOptionsFromVolID(ctx, string(volID), volContext, volSecrets)
+	if err != nil {
+		if !errors.Is(err, cerrors.ErrInvalidVolID) {
+			return nil, err
+		}
+
+		// gets mon IPs from the supplied cluster info
+		volOptions, _, err = core.NewVolumeOptionsFromStaticVolume(string(volID), volContext)
```

Comment:
@gman0 dont we have `clusterID` here from the volContext and just use it with `util.GetMon` or existing function to retriev the monitors?

Reply:
GitHub's in-line comments are a bit weird... Are you referring to [`core.NewVolumeOptionsFromStaticVolume()`](https://github.com/ceph/ceph-csi/blob/0311eb5f447762a581194d649ea4d703da104fb8/internal/cephfs/core/volumeoptions.go#L438-L505)? The `// gets mon IPs from the supplied cluster info` comment is wrong, that function does a lot more than just retrieving mon IPs.

I just copy-pasted what was there before refactoring to minimize changes, but I can change the descriptions because they are indeed a bit misleading.